src/contract.py

```diff
@@ -45,6 +63,12 @@ def get_timeline(self) -> list[float]:
     def _raise_incorrect_derivative_type(self):
         raise TypeError(f'Derivative type of {type(self).__name__} must be CALL or PUT')
 
+    def _raise_incorrect_barrier_updown_type(self):
```

Comment:
These barrier related functions can be moved to `PathDependentContract`.

Reply:
done